File: day05/src/com/oxysa/exercise/Demo03.java
package com.oxysa.exercise;

import com.oxysa.streamtool.Tool;

import java.io.FileInputStream;
import java.io.FileOutputStream;

/**
 * @author Yummy
 * @mail 873387751@qq.com
 * @date 2020-3-17 0:32
 * 案列三：图片加密
 * 核心原理: 一个数字被同一个数字位异或两次, 该数字值不变.
 * 例如:  10 ^ 20 ^ 20 = 10
 * 例如: 5 ^ 10 ^ 5 = 10
 */
public class Demo03 {
    public static void main(String[] args) {
        FileInputStream fis = null;
        FileOutputStream fos = null;


        try {
            fis = new FileInputStream("./day05/data/123.png");
            fos = new FileOutputStream("./day05/data/234.png");
            int len = 0;
            while ((len = fis.read()) != -1) {
                fos.write(len ^ 9);
            }
            System.out.println("加密完成。。。。");
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            Tool.myClose(fis, fos);
        }

    }
}

File: scala_day01/scala/com/oxysa/ifelse/Demo03.scala
package com.oxysa.ifelse

import scala.io.StdIn

/**
  * @author Yummy
  * @mail 873387751@qq.com
  * @date 2020-3-22 18:31
  *       条件控制语句
  *       单分支和双分支
  */
object Demo03 {
  def main(args: Array[String]): Unit = {
    println("请输入第一个数")
    val num1 = StdIn.readInt()
    println("请输入第二个数")
    val num2 = StdIn.readInt()
    if (num1 > 2) {
      if (num2 > 4) {
        println(num1 + num2)
      }
      println("num2不符合条件" + num2)

    } else {
      println("num1 is:" + num1)
    }


  }
}
